### PR TITLE
Payment method options and pagination optimization

### PR DESCRIPTION
**Pull request by cbetta**
* Remove pagination from payment options
* Remove supported currencies/countries from payment options
* Introduce a short-format (`token` format) of a payment method
* Update payment method API to allow for selecting format
* Add list-buyer-payment-methods API to return token-format payment methods for a buyer

### CODE REVIEW

**Comment by douglaseggleton** [comment]
- Retrieving the list of payment options - Does this include both "token" resources and "full" resources? or will this be a separate endpoint I would need to call?

- Payment Method, e.g. Card, Paypal  -  Payment Option, e.g. Tokenized Card, New Card ... is this correct terminology?

File: reference/resources/card--token.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  token:
+    type: string
+    description: |-
+      A Gr4vy token that represents this payment method, in the format of a
+      credit-card number starting with the digit 0. You can store this token
+      in your own system and use it to process a payment with this payment
+      method at a later date.
+    example: "0123456789123458"
+    nullable: true
```
Comment (douglaseggleton) [inline reference/resources/card--token.yaml:24]: Does nullable man a Card Token resource can exist without a token?

**Review by douglaseggleton — APPROVED** [review]

**Review by aliminaei — APPROVED** [review]
Looks good to me

File: reference/resources/card--token.yaml (hunk 1 above)
Comment (cbetta) [reply reference/resources/card--token.yaml:24]: Yes. But thos only affects the list-payment-methods API. In the list-buyer-payment-methods API we will only return tokenized payment methods.

**Pull request merged**